demos/mqtt_transport/mqtt_demo_transport.c

```diff
@@ -0,0 +1,821 @@
+#include <string.h>
+
+/* POSIX socket includes. */
+#include <netdb.h>
+#include <unistd.h>
+#include <assert.h>
+
+#include <sys/socket.h>
+#include <sys/types.h>
+
+
+
+/************ End of logging configuration ****************/
+
+/**
+ * @brief The first characters in the client identifier. A timestamp is appended
+ * to this prefix to create a unique client identifer.
+ *
+ * This prefix is also used to generate topic names and topic filters used in this
+ * demo.
+ */
+#define CLIENT_IDENTIFIER_PREFIX             "iotdemo"
+
+/**
+ * @brief The maximum length the demo will use for client identifier.
+ */
+#define CLIENT_IDENTIFIER_MAX_LENGTH         12U
+
+/**
+ * @brief The topic the demo will subscribe and publish to.
+ */
+#define MQTT_EXAMPLE_TOPIC                   "iotdemo/test/abc"
+
+/**
+ * @brief The size of the buffer in bytes passed to the transport code to hold streaming data.
+ */
+#define STATIC_BUFFER_SIZE                   200U
+
+/**
+ * @brief The MQTT message published in this example.
+ */
+#define MQTT_EXAMPLE_MESSAGE                 "Hello MQTT World!"
+
+/**
+ * @brief Keep alive period in seconds for MQTT connection.
+ */
+#define MQTT_KEEP_ALIVE_PERIOD_SECONDS       5U
+
+/**
+ * @brief Number of time network receive will be attempted
+ * if it fails due to transportTimeout.
+ */
+#define MQTT_MAX_RECV_ATTEMPTS               100U
+
+/**
+ * @brief Delay between two demo iterations.
+ */
+#define MQTT_DEMO_ITERATION_DELAY_SECONDS    5U
+
+#define HIGH_BYTE_MASK                       0xF0U
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Callback for the demo, handles events when the data channel is first opened
+ * and when data is received by the channel.
+ */
+static void demoCallback( IotBleDataTransferChannelEvent_t event,
+                          IotBleDataTransferChannel_t * pChannel,
+                          void * context );
+
+/**
+ * @brief Initializes the demo, including calling the init function for the transport layer
+ * and initializing the BLE channel.
+ */
+static MQTTStatus_t demoInitChannel( void );
+
+
+/**
+ * @brief Generate and return monotonically increasing packet identifier.
+ *
+ * @return The next PacketId.
+ *
+ * @note This function is not thread safe.
+ */
+static uint16_t getNextPacketIdentifier( void );
+
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Packet Identifier generated when Subscribe request was sent to the broker;
+ * it is used to match received Subscribe ACK to the transmitted SUBSCRIBE request.
+ */
+static uint16_t subscribePacketIdentifier;
+
+/**
+ * @brief Packet Identifier generated when Unsubscribe request was sent to the broker;
+ * it is used to match received Unsubscribe response to the transmitted unsubscribe
+ * request.
+ */
+static uint16_t unsubscribePacketIdentifier;
+
+/**
+ * @brief Network context structure to store the data channel.
+ */
+static NetworkContext_t pContext;
```

Comment:
Prefix 'p' suggests a pointer. You can either prefix with 'x' for a structure or leave it without 'p' to avoid confusion.